eth/handler.go

```diff
@@ -234,10 +234,13 @@ func newHandler(config *handlerConfig) (*handler, error) {
 func (h *handler) runEthPeer(peer *eth.Peer, handler eth.Handler) error {
 	// If the peer has a `snap` extension, wait for it to connect so we can have
 	// a uniform initialization/teardown mechanism
-	snap, err := h.peers.waitSnapExtension(peer)
-	if err != nil {
-		peer.Log().Error("Snapshot extension barrier failed", "err", err)
-		return err
+	var snapPeer *snap.Peer
+	if peer.RunningCap(snap.ProtocolName, snap.ProtocolVersions) {
```

Comment:
This check is not necessary here. The original code is fine, will return `nil` if snap is not enabled. No need to check it here too.

Reply:
Ah right -- I added this first, and later changed the inner one. and didn't consider that the inner change made this moot